### ISSUE TITLE
Lab 07 Cannot compile the the function  

### ISSUE THREAD

**Issue by tibor19** [MicrosoftLearning/AZ-204-DevelopingSolutionsforMicrosoftAzure#72]
# Module: 07
#### Step: 04

Description of issue
When the student presses Test run the function fails with an Access Denied error
Repro steps:

1. Run the lab
2. Press Run
3. See that it doesn't work :)

**Comment by seesharprun** [comment]
Can you share a screenshot of where you see the **Access Denied** error message?

**Comment by tibor19** [comment]
MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
System.ComponentModel.Win32Exception (5): Access is denied.
   at System.Diagnostics.Process.set_PriorityClassCore(ProcessPriorityClass value)
   at System.Diagnostics.Process.set_PriorityClass(ProcessPriorityClass value)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String[] commandLine)
Unhandled exception. System.ComponentModel.Win32Exception (5): Access is denied.
   at System.Diagnostics.Process.set_PriorityClassCore(ProcessPriorityClass value)
   at System.Diagnostics.Process.set_PriorityClass(ProcessPriorityClass value)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String[] commandLine)
   at Microsoft.Build.CommandLine.MSBuildApp.Main(String[] args)

**Comment by Camios** [comment]
I'm having a problem too.
I see (why does the azure code + test "logs" window have duplicates?)
```
2020-07-01T04:58:39Z   [Information]   Executing 'Functions.FileParser' (Reason='This function was programmatically called via the host APIs.', Id=f095ac11-5013-413f-ae08-40943368177a)
2020-07-01T04:58:39Z   [Information]   Package references have been updated.
2020-07-01T04:58:39Z   [Information]   Restoring packages.
2020-07-01T04:58:39Z   [Information]   Starting packages restore
2020-07-01T04:58:41Z   [Information]   MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
2020-07-01T04:58:41Z   [Information]   System.ComponentModel.Win32Exception (5): Access is denied.
2020-07-01T04:58:41Z   [Information]      at System.Diagnostics.Process.set_PriorityClassCore(ProcessPriorityClass value)
2020-07-01T04:58:41Z   [Information]      at System.Diagnostics.Process.set_PriorityClass(ProcessPriorityClass value)
2020-07-01T04:58:41Z   [Information]      at Microsoft.Build.CommandLine.MSBuildApp.Execute(String[] commandLine)
2020-07-01T04:58:41Z   [Information]   Unhandled exception. System.ComponentModel.Win32Exception (5): Access is denied.
2020-07-01T04:58:41Z   [Information]      at System.Diagnostics.Process.set_PriorityClassCore(ProcessPriorityClass value)
2020-07-01T04:58:41Z   [Information]      at System.Diagnostics.Process.set_PriorityClass(ProcessPriorityClass value)
2020-07-01T04:58:41Z   [Information]      at Microsoft.Build.CommandLine.MSBuildApp.Execute(String[] commandLine)
2020-07-01T04:58:41Z   [Information]      at Microsoft.Build.CommandLine.MSBuildApp.Main(String[] args)
2020-07-01T04:58:41Z   [Information]   Packages restored.
2020-07-01T04:58:42Z   [Warning]   You may be referencing NuGet packages incorrectly. Learn more: https://go.microsoft.com/fwlink/?linkid=2091419
2020-07-01T04:58:42Z   [Warning]   You may be referencing NuGet packages incorrectly. Learn more: https://go.microsoft.com/fwlink/?linkid=2091419
2020-07-01T04:58:42Z   [Warning]   You may be referencing NuGet packages incorrectly. Learn more: https://go.microsoft.com/fwlink/?linkid=2091419
2020-07-01T04:58:42Z   [Warning]   You may be referencing NuGet packages incorrectly. Learn more: https://go.microsoft.com/fwlink/?linkid=2091419
2020-07-01T04:58:42Z   [Warning]   You may be referencing NuGet packages incorrectly. Learn more: https://go.microsoft.com/fwlink/?linkid=2091419
2020-07-01T04:58:42Z   [Warning]   You may be referencing NuGet packages incorrectly. Learn more: https://go.microsoft.com/fwlink/?linkid=2091419
2020-07-01T04:58:42Z   [Warning]   You may be referencing NuGet packages incorrectly. Learn more: https://go.microsoft.com/fwlink/?linkid=2091419
2020-07-01T04:58:42Z   [Error]   run.csx(3,7): error CS0246: The type or namespace name 'Azure' could not be found (are you missing a using directive or an assembly reference?)
2020-07-01T04:58:42Z   [Error]   run.csx(4,7): error CS0246: The type or namespace name 'Azure' could not be found (are you missing a using directive or an assembly reference?)
2020-07-01T04:58:42Z   [Error]   run.csx(5,7): error CS0246: The type or namespace name 'Azure' could not be found (are you missing a using directive or an assembly reference?)
2020-07-01T04:58:42Z   [Error]   run.csx(10,5): error CS0246: The type or namespace name 'BlobServiceClient' could not be found (are you missing a using directive or an assembly reference?)
2020-07-01T04:58:42Z   [Error]   run.csx(10,5): error CS0246: The type or namespace name 'BlobServiceClient' could not be found (are you missing a using directive or an assembly reference?)
```

**Comment by ecaha** [comment]
The issue was there - resolved by appsetting DOTNET_ADD_GLOBAL_TOOLS_TO_PATH=false. I think is back again, alive and kicking. Only workaround which I was able to find - develop azure function in VSCode/VS and deploy.

**Comment by seesharprun** [comment]
Long term, we will have to rewrite the lab to develop locally and deploy.

Short term, we will try and add the app setting back to the lab.

**Comment by seesharprun** [comment]
This is now resolved in e4cd471.

The short term solution is to set the configuration settings like this:

| Configuration Setting | Value |
|--- | ---|
| ``DOTNET_ADD_GLOBAL_TOOLS_TO_PATH`` | ``false`` |
| ``DOTNET_SKIP_FIRST_TIME_EXPERIENCE`` | ``true`` |

The issues is discussed at [azure-functions-host#5893](https://github.com/Azure/azure-functions-host/issues/5893). It's still technically unresolved.
